src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/TempDirectoryWorkloadManifestProvider.cs

```diff
@@ -4,27 +4,47 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Linq;
 
 namespace Microsoft.NET.Sdk.WorkloadManifestReader
 {
     public class TempDirectoryWorkloadManifestProvider : IWorkloadManifestProvider
     {
-        private readonly string _sdkVersionBand;
         private readonly string _manifestsPath;
+        private readonly string _sdkVersionBand;
 
+        /// <param name="manifestsPath">
+        ///     Result of directly extract manifest NuGet Packages. Should contain folders like
+        ///     microsoft.net.workload.emscripten.manifest-6.0.100.6.0.0-preview.7.21377.2
+        /// </param>
         public TempDirectoryWorkloadManifestProvider(string manifestsPath, string sdkVersion)
         {
             _manifestsPath = manifestsPath;
             _sdkVersionBand = sdkVersion;
         }
 
-        public IEnumerable<(string manifestId, string? informationalPath, Func<Stream> openManifestStream)> GetManifests()
+        public IEnumerable<(string manifestId, string? informationalPath, Func<Stream> openManifestStream)>
+            GetManifests()
         {
             foreach (var workloadManifestDirectory in GetManifestDirectories())
             {
-                var workloadManifest = Path.Combine(workloadManifestDirectory, "WorkloadManifest.json");
-                var id = Path.GetFileName(workloadManifestDirectory);
-                yield return (id, workloadManifest, () => File.OpenRead(workloadManifest));
+                string? workloadManifest = Path.Combine(workloadManifestDirectory, "WorkloadManifest.json");
+                FileInfo? nupkg = new DirectoryInfo(workloadManifestDirectory).GetFiles("*.nuspec",
+                    SearchOption.TopDirectoryOnly).SingleOrDefault();
+                if (nupkg == default)
+                {
+                    throw new Exception($"Package {nupkg} contains more than one nuspec");
+                }
+
+                string? packageId = Path.GetFileNameWithoutExtension(nupkg.FullName);
+
+                int index = packageId.IndexOf(".Manifest", StringComparison.OrdinalIgnoreCase);
+                if (index >= 0)
+                {
+                    packageId = packageId.Substring(0, index);
```

Comment:
For clarity, can we put this in a separate var called `manifestId` since the package id would be `microsoft.net.workload.emscripten.manifest-6.0.100` but we have the manifest id here, which is `microsoft.net.workload.emscripten`?

Reply:
changed to manifestId